patterns/grok-patterns

```diff
@@ -35,7 +35,7 @@ HOSTPORT %{IPORHOST}:%{POSINT}
 
 # paths
 PATH (?:%{UNIXPATH}|%{WINPATH})
-UNIXPATH (/([\w_%!$@:.,+~-]+|\\.)*)+
+UNIXPATH (?>\/[[\w_%!$@:.,-]+|\\.]*)+
```

Comment:
I'm not sure I'm reading this correctly. Can you help me understand?

I see two main changes. 

One, using an atomic group. I like. 

Two, it seems like the character class `[....]` was wrapped with another `[...]` and includes the `+|\\.` bits, which feels incorrect to me.

Here's a difference of before/after. It looks like the "after" is wrong. It shouldn't be one big charater class. Should it?

```diff
-[\w_%!$@:.,+~-]+|\\.
+[[\w_%!$@:.,-]+|\\.]
```